sdks/java/io/kafka/src/main/java/org/apache/beam/sdk/io/kafka/KafkaUnboundedReader.java

```diff
@@ -299,6 +303,30 @@ public Instant getCurrentTimestamp() throws NoSuchElementException {
     return curTimestamp;
   }
 
+  @Override
+  public byte[] getCurrentRecordId() throws NoSuchElementException {
+    if (curId == null) {
+      if (this.offsetDeduplication) {
+        throw new NoSuchElementException();
+      } else {
+        return new byte[0];
+      }
+    }
+    return curId;
+  }
+
+  @Override
+  public byte[] getCurrentRecordOffset() throws NoSuchElementException {
```

Comment:
ditto

Reply:
Responded above but leaving this open to make sure any subsequent changes get reflected here too.